Windows 365 Gateway IP Lookup/W365 Gateway IP Address Retrieval.ps1

```diff
@@ -0,0 +1,151 @@
+﻿<#
+.COPYRIGHT
+Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+See LICENSE in the project root for license information.
+#>
+
+#v1.0
+
+#output CSV file
+$CSVFile = "c:\temp\W365-Gateways.CSV"
+
+#modules required for this script
+$modules = @("az.accounts",
+    "az.network"
+)
+
+#function to ensure modules are installed and current
+function invoke-modulecheck {
+    foreach ($module in $modules) {
+
+        write-host "Checking module - " $module
+        $graphcurrent = (get-installedmodule -name $module -ErrorAction SilentlyContinue)
+        if ($graphcurrent -eq $null) {
+            write-output "Module is not installed. Installing..." | out-host
+            try {
+                Install-Module -name $module -Force -ErrorAction Stop 
+                Import-Module -name $module -force -ErrorAction Stop 
+
+            }
+            catch {
+                write-output "Failed to install " $module | out-host
+                write-output $_.Exception.Message | out-host
+                Exit
+            }
+        }
+
+        $currentver = (find-module -name $module).version.ToString()
+        $installedver = (get-installedmodule -name $module).version.tostring()
+
+        if ($currentver -gt $installedver) { write-host "There is an update to this module." }
+        else
+        { write-host "This module is up to date" }
+
+    }
+}
+
+#prompt user to connect to Azure
+function connect-azure {
+
+    $tenant = get-aztenant -ErrorAction SilentlyContinue
+    if ($tenant.TenantId -eq $null) {
+        write-output "Not connected to Azure. Connecting..." | out-host
+        try {
+            Connect-AzAccount -ErrorAction Stop | Out-Null
+        }
+        catch {
+            write-output "Failed to connect to Azure" | out-host
+            write-output $_.Exception.Message | out-host
+            Exit
+        }   
+    }
+    $tenant = get-aztenant
+    $text = "Tenant ID is " + $tenant.TenantId
```

Comment:
What as if the tenant Id is still null, due to some accidence? May be you'd like to have one more check here and exit the script if the tenant Id is still null.

Reply:
$tenant = get-aztenant
    if ($tenant.Tenant.ID -eq $null){
        write-output "Couldn't retrieve the tenant id. Try again." | out-host
        write-output $_.Exception.Message | out-host
        Exit
    }